### PR TITLE
Adding support for DiagnosticServer Build

### PR DESCRIPTION
**Pull request by kumaramit-msft**

### CODE REVIEW

File: BuildAndTagImages/build.sh [hunk 1]
```diff
@@ -87,6 +87,33 @@ function buildDockerImage()
...
+        ls "${SYSTEM_ARTIFACTS_DIR}"
+        echo "Listing stacks dir"
+        cd "${SYSTEM_ARTIFACTS_DIR}/${STACK}/${FILES_ROOT_PATH}"
+        echo
+        echo "Building test image with tag '$BuildVerRepoTag' and file $appSvcDockerfilePath..."
+        echo docker build -t "$BuildVerRepoTag" -f "$appSvcDockerfilePath" .
+        docker build -t "$BuildVerRepoTag" -f "$appSvcDockerfilePath" .
+        docker tag $BuildVerRepoTag $MCRRepoTag
+
+        # only push the images if merging to the master
```
Comment (rahulaga-msft) [inline BuildAndTagImages/build.sh:109]: am not sure if this comment still holds good here - I believe you intent to create image out of `main` branch. please verify

Comment (kumaramit-msft) [reply BuildAndTagImages/build.sh:109]: Yes, will update it to say main!

File: GenerateDockerFiles/DiagnosticServer/generateDockerfiles.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+declare -r APPSVC_DIAGSERVER_REPO="$2"
+declare -r CONFIG_DIR="$3"                                         # ${Current_Repo}/Config
+declare -r METADATA_FILE="$SYSTEM_ARTIFACTS_DIR/metadata"
+declare -r APP_SVC_REPO_DIR="$SYSTEM_ARTIFACTS_DIR/$STACK_NAME/GitRepo"
+declare -r APP_SVC_REPO_BRANCH="main"
+
+function generateDiagnosticServerDockerFiles()
+{
+    CURR_VERSION_DIRECTORY="${APP_SVC_REPO_DIR}/"
+    TARGET_DOCKERFILE="${CURR_VERSION_DIRECTORY}/Dockerfile"
```
Comment (rahulaga-msft) [inline GenerateDockerFiles/DiagnosticServer/generateDockerfiles.sh:25]: essentially we are using dockerFile from repo as it is - can this have any limitation in terms of parametrizing anything going forward ? please evaluate.

**Review by rahulaga-msft — APPROVED** [review]
looks good to me

File: GenerateDockerFiles/DiagnosticServer/generateDockerfiles.sh (hunk 2 above)
Comment (kumaramit-msft) [reply GenerateDockerFiles/DiagnosticServer/generateDockerfiles.sh:25]: Thats a good suggestion, not kept it parameterized as we also build it locally in Diagnostic Server Pipeline.

**Review by purva-vasudeo — APPROVED** [review]

**Pull request merged**